### PR TITLE
Drop ruby 2.6 and 2.7 support

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
## What was the end-user or developer problem that led to this PR?

Time to move on. These are EOL.

## What is your fix for the problem, implemented in this PR?

Delete all the things.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

File: lib/rubygems/basic_specification.rb [hunk 1]
```diff
@@ -133,9 +131,9 @@ def full_gem_path
 
   def full_name
     if platform == Gem::Platform::RUBY || platform.nil?
-      (+"#{name}-#{version}").tap(&Gem::UNTAINT)
```
Comment (segiddins) [inline lib/rubygems/basic_specification.rb:136]: i think these can now safely remove the `+`

File: lib/rubygems/basic_specification.rb [hunk 2]
```diff
@@ -295,7 +293,7 @@ def lib_dirs_glob
       "lib" # default value for require_paths for bundler/inline
     end
 
-    "#{full_gem_path}/#{dirs}".dup.tap(&Gem::UNTAINT)
+    "#{full_gem_path}/#{dirs}".dup
```
Comment (segiddins) [inline lib/rubygems/basic_specification.rb:296]: remove `.dup`

File: lib/rubygems/bundler_version_finder.rb [hunk 3]
```diff
@@ -67,7 +67,7 @@ def self.lockfile_contents
     lockfile = case gemfile
                when "gems.rb" then "gems.locked"
                else "#{gemfile}.lock"
-    end.dup.tap(&Gem::UNTAINT)
+    end.dup
```
Comment (segiddins) [inline lib/rubygems/bundler_version_finder.rb:70]: remove `.dup`

File: lib/rubygems/config_file.rb [hunk 4]
```diff
@@ -188,7 +188,7 @@ def initialize(args)
     operating_system_config = Marshal.load Marshal.dump(OPERATING_SYSTEM_DEFAULTS)
     platform_config = Marshal.load Marshal.dump(PLATFORM_DEFAULTS)
     system_config = load_file SYSTEM_WIDE_CONFIG_FILE
-    user_config = load_file config_file_name.dup.tap(&Gem::UNTAINT)
+    user_config = load_file config_file_name.dup
```
Comment (segiddins) [inline lib/rubygems/config_file.rb:191]: remove `.dup`

File: lib/rubygems/core_ext/kernel_require.rb [hunk 5]
```diff
@@ -46,7 +46,7 @@ def require(path) # :doc:
           load_path_check_index = Gem.load_path_insert_index - Gem.activated_gem_paths
           Gem.suffixes.find do |s|
             $LOAD_PATH[0...load_path_check_index].find do |lp|
-              safe_lp = lp.dup.tap(&Gem::UNTAINT)
+              safe_lp = lp.dup
```
Comment (segiddins) [inline lib/rubygems/core_ext/kernel_require.rb:49]: remove `.dup`

File: lib/rubygems/name_tuple.rb [hunk 6]
```diff
@@ -51,7 +51,7 @@ def full_name
       "#{@name}-#{@version}"
     else
       "#{@name}-#{@version}-#{@platform}"
-    end.dup.tap(&Gem::UNTAINT)
+    end.dup
```
Comment (segiddins) [inline lib/rubygems/name_tuple.rb:54]: remove `.dup`

File: lib/rubygems/path_support.rb [hunk 7]
```diff
@@ -48,7 +48,7 @@ def initialize(env)
 
     @spec_cache_dir = env["GEM_SPEC_CACHE"] || Gem.default_spec_cache_dir
 
-    @spec_cache_dir = @spec_cache_dir.dup.tap(&Gem::UNTAINT)
+    @spec_cache_dir = @spec_cache_dir.dup
```
Comment (segiddins) [inline lib/rubygems/path_support.rb:51]: remove `.dup`

File: lib/rubygems/request_set.rb [hunk 8]
```diff
@@ -324,7 +324,7 @@ def load_gemdeps(path, without_groups = [], installing = false)
 
     @git_set.root_dir = @install_dir
 
-    lock_file = "#{File.expand_path(path)}.lock".dup.tap(&Gem::UNTAINT)
+    lock_file = "#{File.expand_path(path)}.lock".dup
```
Comment (segiddins) [inline lib/rubygems/request_set.rb:327]: remove `.dup`

File: lib/rubygems/specification.rb [hunk 9]
```diff
@@ -961,7 +961,7 @@ def self.attribute_names
 
   def self.dirs
     @@dirs ||= Gem.path.collect do |dir|
-      File.join dir.dup.tap(&Gem::UNTAINT), "specifications"
+      File.join dir.dup, "specifications"
```
Comment (segiddins) [inline lib/rubygems/specification.rb:964]: remove `.dup`

File: lib/rubygems/specification.rb [hunk 10]
```diff
@@ -1154,13 +1154,11 @@ def self.load(file)
     spec = @load_cache_mutex.synchronize { @load_cache[file] }
     return spec if spec
 
-    file = file.dup.tap(&Gem::UNTAINT)
+    file = file.dup
```
Comment (segiddins) [inline lib/rubygems/specification.rb:1157]: remove line entirely

**Review by segiddins — APPROVED** [review]
beautiful!!

File: bundler/bundler.gemspec [hunk 11]
```diff
@@ -29,8 +29,8 @@ Gem::Specification.new do |s|
     "source_code_uri" => "https://github.com/rubygems/rubygems/tree/master/bundler",
   }
 
-  s.required_ruby_version     = ">= 2.6.0"
-  s.required_rubygems_version = ">= 3.0.1"
+  s.required_ruby_version     = ">= 3.0.0"
+  s.required_rubygems_version = ">= 3.1.2"
```
[the remark this answers is not in the document]
Comment (deivid-rodriguez) [reply bundler/bundler.gemspec:33]: Yes, I will add a note!

File: lib/rubygems/commands/update_command.rb [hunk 12]
```diff
@@ -328,12 +328,8 @@ def oldest_supported_version
     @oldest_supported_version ||=
       if Gem.ruby_version > Gem::Version.new("3.1.a")
         Gem::Version.new("3.3.3")
-      elsif Gem.ruby_version > Gem::Version.new("3.0.a")
-        Gem::Version.new("3.2.3")
-      elsif Gem.ruby_version > Gem::Version.new("2.7.a")
-        Gem::Version.new("3.1.2")
       else
-        Gem::Version.new("3.0.1")
+        Gem::Version.new("3.2.3")
```
Comment (martinemde) [inline lib/rubygems/commands/update_command.rb:332]: Above you said 3.1.0 shipped with ruby 3.0, but was it actually 3.2.3?

**Review by martinemde — APPROVED** [review]
Beautiful! 🤩

**Review by hsbt — APPROVED** [review]
:+1: Great work!

File: lib/rubygems/commands/update_command.rb (hunk 12 above)
Comment (deivid-rodriguez) [reply lib/rubygems/commands/update_command.rb:332]: Yes, my bad, this is correct, the gemspec info is wrong. Will correct it.

**Review by simi — APPROVED** [review]
finally :pray:

**Comment by nobu** [comment]
Also `Gem::NoAliasYAMLTree.create` and `Gem::NoAliasYAMLTree#format_time` can be removed, I think.

**Review by duckinator — APPROVED** [review]
This is fantastic. I reviewed all the non-deleted files and everything looks good to me. I didn't manually inspect the deleted files, but they appear to entirely consist of `tool/bundler/*26_gems.rb*`, `.github/workflows/legacy-git.yml`, and over 1,300 cached network requests under `vcr_cassettes` &mdash; and I don't think that should give us problems if CI passes for this. 🙂

**Comment by deivid-rodriguez** [comment]
Yeah! I was a bit disappointed that the work I did at https://github.com/rubygems/rubygems/pull/6208 never paid off, but it finally showed up to remind us to delete 1.3k files 😃

**Comment by deivid-rodriguez** [comment]
Let's do it!

**Pull request merged**